solana/pyth2wormhole/client/src/batch_state.rs

```diff
@@ -40,63 +42,59 @@ impl<'a> BatchState<'a> {
             symbols,
             conditions,
             last_known_symbol_states: vec![None; symbols.len()],
-            status: BatchTxStatus::Sending { attempt_no: 1 },
-            status_changed_at: Instant::now(),
-            last_success_at: None,
+            last_job_finished_at: Instant::now(),
         }
     }
-    /// Ensure only set_status() alters the timestamp
-    pub fn get_status_changed_at(&self) -> &Instant {
-        &self.status_changed_at
-    }
-    pub fn get_status(&self) -> &BatchTxStatus {
-        &self.status
-    }
-
-    /// Ensure that status changes are accompanied by a timestamp bump
-    pub fn set_status(&mut self, s: BatchTxStatus) {
-        self.status_changed_at = Instant::now();
-        self.status = s;
-    }
 
     /// Evaluate the configured attestation conditions for this
     /// batch. RPC is used to update last known state. Returns
     /// Some("<reason>") if any trigger condition was met. Only the
     /// first encountered condition is mentioned.
-    pub fn should_resend(&mut self, c: &RpcClient) -> Option<String> {
+    pub async fn should_resend(&mut self, c: &RpcClient) -> Option<String> {
         let mut ret = None;
 
         let sym_count = self.symbols.len();
         let mut new_symbol_states: Vec<Option<PriceAccount>> = Vec::with_capacity(sym_count);
+        let mut lookup_failures = vec![];
         for (idx, sym) in self.symbols.iter().enumerate() {
             let new_state = match c
                 .get_account_data(&sym.price_addr)
                 .map_err(|e| e.to_string())
-                .and_then(|bytes| {
+                .and_then(|bytes| async move {
                     pyth_sdk_solana::state::load_price_account(&bytes)
                         .map(|state| state.clone())
                         .map_err(|e| e.to_string())
-                }) {
+                })
+                .await
```

Comment:
You are correct, I intentionally only refactored the core event loop so that only essential complexity is introduced. The account state lookups are reasonably quick and I don't consider them important to parallelize. They're already concurrent across batches thanks to the main.rs changes